web/src/compose_validate.js

```diff
@@ -383,6 +385,33 @@ function is_recipient_large_stream() {
     );
 }
 
+export function topic_participant_count_more_than_threshold(stream_id, topic) {
+    // Topic participants:
+    // Users who either sent or reacted to the messages in the topic.
+    const participant_ids = new Set();
+    const topic_messages = message_util.get_messages_in_topic(stream_id, topic);
+
+    for (const mssg of topic_messages) {
+        participant_ids.add(mssg.sender_id);
```

Comment:
I think this would be expensive as `get_messages_in_topic` iterates over `all_messages_data.all_messages()` to find topic_messages.

Reply:
Cool, `recent_senders.get_topic_recent_senders` was helpful.